features/268-next-cf/api.md

```diff
@@ -614,3 +614,53 @@ Currently, micro-service allow developers to do routing in the app, while others
 - Custom Domains Support #246 (https://github.com/SkygearIO/features/issues/246)
 
 Handled by the k8s ingress controller. Skycli may provide command for skygear user to add these features.
+
+## Secret management
+
+With immutable deployment, secrets are also immutable. For managing secrets,
+following operations are allowed.
+
+- skycli secret create
+- skycli secret remove
+- skycli secret rename
+
+To avoid human error, deployment will be failed if config file is using secrets
+that are not exist. Removing or renaming a secret will not affect existing deployment.
+If skygear user want to update a secret (e.g. key rotation), he will need to rename the
+existing secret to another name. Create a secret with the same name and re-deploy the functions.
+
+### Example: Key rotation
+
+```sh
+$ skycli secret rename MY_SECRET_KEY MY_SECRET_KEY_OLD
```

Comment:
Will we provide a function to rotate key in future? A recommended way to handle it as a separate feature issue?